plugins/in_syslog/syslog_prot.c

```diff
@@ -31,88 +33,127 @@ static inline void consume_bytes(char *buf, int bytes, int length)
+            if (result == 0) {
+                result = FLB_MAP_EXPAND_SUCCESS;
+            }
+            else {
+                result = FLB_MAP_EXPANSION_ERROR;
+            }
+        }
     }
 
-    msgpack_sbuffer_init(&mp_sbuf);
-    msgpack_packer_init(&mp_pck, &mp_sbuf, msgpack_sbuffer_write);
-
-    msgpack_pack_map(&mp_pck, unpacked_buffer.data.via.map.size + 1);
-
-    for (i = 0; i < unpacked_buffer.data.via.map.size; i++) {
-        msgpack_pack_object(&mp_pck, unpacked_buffer.data.via.map.ptr[i].key);
-        msgpack_pack_object(&mp_pck, unpacked_buffer.data.via.map.ptr[i].val);
+    if (result == FLB_MAP_EXPAND_SUCCESS) {
+        *result_buffer = modified_data_buffer;
+        *result_size = modified_data_size;
     }
 
-    msgpack_pack_str(&mp_pck, flb_sds_len(raw_message_key_name));
-    msgpack_pack_str_body(&mp_pck, raw_message_key_name, flb_sds_len(raw_message_key_name));
-    msgpack_pack_str(&mp_pck, raw_message_size);
-    msgpack_pack_str_body(&mp_pck, raw_message_buffer, raw_message_size);
-
-    *result_buffer = mp_sbuf.data;
-    *result_size = mp_sbuf.size;
-
-    msgpack_unpacked_destroy(&unpacked_buffer);
     return result;
 }
 
 static inline int pack_line(struct flb_syslog *ctx,
                             struct flb_time *time,
+                            struct flb_connection *connection,
                             char *data, size_t data_size,
                             char *raw_data, size_t raw_data_size)
 {
     char   *modified_data_buffer;
     size_t  modified_data_size;
+    char   *appended_address_buffer;
+    size_t  appended_address_size;
     int     result;
+    char   *source_address;
 
+    source_address = NULL;
     modified_data_buffer = NULL;
+    appended_address_buffer = NULL;
 
     if (ctx->raw_message_key != NULL) {
-        result = append_raw_message_to_record_data(&modified_data_buffer,
-                                                   &modified_data_size,
-                                                   ctx->raw_message_key,
-                                                   data,
-                                                   data_size,
-                                                   raw_data,
-                                                   raw_data_size);
-
-        if (result != 0) {
-            flb_plg_debug(ctx->ins, "error appending raw message : %d", result);
+        result = append_message_to_record_data(&modified_data_buffer,
+                                               &modified_data_size,
+                                               ctx->raw_message_key,
+                                               data,
+                                               data_size,
+                                               raw_data,
+                                               raw_data_size,
+                                               MSGPACK_OBJECT_BIN);
+
+        if (result == FLB_MAP_EXPANSION_ERROR) {
```

Comment:
`if (result != FLB_MAP_EXPAND_SUCCESS) {` otherwise, you are printing a value you already since the inner result code is lost.